File: rust/507.perfect-number.rs
/*
 * @lc app=leetcode id=507 lang=rust
 *
 * [507] Perfect Number
 */

// @lc code=start
impl Solution {
    pub fn check_perfect_number(num: i32) -> bool {
        if num % 2 != 0 {
            return false;
        }
        let mut perfect_number = 0;
        for i in 2..=(num as f64).sqrt() as i32 {
            if num % i == 0 {
                perfect_number += i + (num / i);
            }
        }
        perfect_number + 1 == num
    }
}
// @lc code=end

File: javascript/507.perfect-number.js
/*
 * @lc app=leetcode id=507 lang=javascript
 *
 * [507] Perfect Number
 */

// @lc code=start
/**
 * @param {number} num
 * @return {boolean}
 */
var checkPerfectNumber = function (num) {
  if (num % 2 !== 0) return false;

  let perfectNumber = 0;
  for (let i = 2; i <= ~~Math.sqrt(num); i++) {
    if (num % i === 0) perfectNumber += i + num / i;
  }
  return perfectNumber + 1 === num;
};
// @lc code=end
